bitnami/kafka/templates/svc-headless.yaml

```diff
@@ -4,7 +4,7 @@ metadata:
   name: {{ printf "%s-headless" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" }}
   namespace: {{ .Release.Namespace | quote }}
   labels: {{- include "common.labels.standard" . | nindent 4 }}
-    app.kubernetes.io/component: kafka
+    app.kubernetes.io/component: kafka-headless
```

Comment:
Yes, that would work also. Leave the component and add a new annotation to both with distinct values on each. Do you have a preference on that annotation name?

Perhaps instance? 
https://kubernetes.io/docs/reference/labels-annotations-taints/#app-kubernetes-io-instance

Reply:
Interestingly there is a well-known haedless label as well:
https://kubernetes.io/docs/reference/labels-annotations-taints/#servicekubernetesioheadless

Reply:
I have been briefly looking into this and it seems that the `service.kubernetes.io/headless` label you mention is automatically added to the Endpoints that are part of a headless service:

```bash
$ k describe endpoints kafka-headless
Name:         kafka-headless
Namespace:    default
Labels:       app.kubernetes.io/component=kafka
              app.kubernetes.io/instance=kafka
              app.kubernetes.io/managed-by=Helm
              app.kubernetes.io/name=kafka
              helm.sh/chart=kafka-16.2.9
              service.kubernetes.io/headless=
```

So, as the description in the documentation says, it is actually a label that is added by the control plane itself. If someone wants to specify a different value other than `""`, we can do it in the SVC. Hence, I don't think this is quite the use case we want.

Instead, I think it would be more useful if we added the possibility of including `extraLabels` for the `headless` service, similar to what we do in `bitnami/mariadb-galera` for annotations. This is more flexible and allows anyone to add new labels as needed, which I believe is the right way to proceed here.

https://github.com/bitnami/charts/blob/66d023a699b718ea2d4fbf616f5cc28e0c1a7d77/bitnami/mariadb-galera/values.yaml#L149-L153

Reply:
I follow that will update when I can

Reply:
Looking neat! The new changes do not justify a MAJOR, a MINOR would suffice :)